libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/ProcessPreBuiltCohort.ts

```diff
@@ -0,0 +1,235 @@
+              arg: preBuiltCohortFilter.arg,
+              column: JointDataset.IndexLabel,
+              method: preBuiltCohortFilter.method
+            } as IFilter;
+            filterList.push(filter);
+            break;
+          }
+          case CohortColumnNames.RegressionError: {
+            const filter: IFilter = {
+              arg: preBuiltCohortFilter.arg,
+              column: JointDataset.RegressionError,
+              method: preBuiltCohortFilter.method
+            } as IFilter;
+            filterList.push(filter);
+            break;
+          }
+          default: {
+            const [filter, errorString] =
+              translatePreBuiltCohortFilterForDataset(
+                preBuiltCohortFilter,
+                jointDataset
+              );
+            if (filter !== undefined) {
+              filterList.push(filter);
+            } else if (errorString !== undefined) {
+              errorStrings.push(errorString);
+            }
+
+            break;
+          }
+        }
+      }
+      const errorCohortEntry = new ErrorCohort(
+        new Cohort(preBuiltCohort.name, jointDataset, filterList),
+        jointDataset
+      );
+      errorCohortList.push(errorCohortEntry);
+    }
+  }
+  return [errorCohortList, errorStrings];
+}
+
+function translatePreBuiltCohortFilterForTarget(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset,
+  cohortColumnName: CohortColumnNames
+): IFilter {
+  let filterColumnName = JointDataset.PredictedYLabel;
+  if (cohortColumnName === CohortColumnNames.TrueY) {
+    filterColumnName = JointDataset.TrueYLabel;
+  }
+  if (
+    jointDataset.getModelType() === ModelTypes.Multiclass ||
+    jointDataset.getModelType() === ModelTypes.Binary
+  ) {
+    const modelClasses = jointDataset.getModelClasses();
+    const index: number[] = [];
+    for (const modelClass of preBuiltCohortFilter.arg) {
+      const indexModelClass = modelClasses.indexOf(modelClass);
+
+      if (indexModelClass !== -1) {
+        index.push(indexModelClass);
+      }
+    }
+
+    index.sort((a, b) => a - b);
+
+    const filter: IFilter = {
+      arg: index,
+      column: filterColumnName,
+      method: preBuiltCohortFilter.method
+    } as IFilter;
+    return filter;
+  }
+  const filter: IFilter = {
+    arg: preBuiltCohortFilter.arg,
+    column: filterColumnName,
+    method: preBuiltCohortFilter.method
+  } as IFilter;
+  return filter;
+}
+
+function translatePreBuiltCohortFilterForClassificationOutcome(
+  preBuiltCohortFilter: IPreBuiltFilter,
+  jointDataset: JointDataset
+): IFilter {
+  const index: number[] = [];
+  if (jointDataset.metaDict[JointDataset.ClassificationError]) {
+    const allowedCalssificationErrorValues =
+      jointDataset.metaDict[JointDataset.ClassificationError]
+        .sortedCategoricalValues;
+
+    if (allowedCalssificationErrorValues !== undefined) {
+      for (const classificationError of preBuiltCohortFilter.arg) {
```

Comment:
shouldn't this be a forEach?  I might be missing something.  I think that's the more functional style that's preferred with typescript.  I could be wrong here though.